File: aidlclient/src/main/java/cn/isif/aidlservice/Book.java
package cn.isif.aidlservice;

import android.os.Parcel;
import android.os.Parcelable;

import androidx.annotation.NonNull;

/**
 * 要想在Binder中传输对象，该对象必须实现Parcelable
 */
public class Book implements Parcelable {
    public String bookName;
    public double price;

    public Book(double price, String bookName) {
        this.price = price;
        this.bookName = bookName;
    }

    protected Book(Parcel in) {
        price = in.readDouble();
        bookName = in.readString();
    }

    public static final Creator<Book> CREATOR = new Creator<Book>() {
        @Override
        public Book createFromParcel(Parcel in) {
            return new Book(in);
        }

        @Override
        public Book[] newArray(int size) {
            return new Book[size];
        }
    };

    public Book() {

    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeDouble(price);
        dest.writeString(bookName);
    }

    public void readFromParcel(Parcel reply) {
        price = reply.readDouble();
        bookName = reply.readString();
    }

    @NonNull
    @Override
    public String toString() {
        return "[name: " + bookName + ", price: " + price + "]";
    }
}

File: aidl/src/main/java/cn/isif/aidl/Book.kt
package cn.isif.aidl

import android.os.Parcel
import android.os.Parcelable

class Book(var name: String? = "", var price:Double = 0.0) :Parcelable{
    constructor(parcel: Parcel) : this(
            parcel.readString(),
            parcel.readDouble()) {
    }

    override fun writeToParcel(dest: Parcel?, flags: Int) {
        dest?.writeString(name)
        dest?.writeDouble(price)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Book> {
        override fun createFromParcel(parcel: Parcel): Book {
            return Book(parcel)
        }

        override fun newArray(size: Int): Array<Book?> {
            return arrayOfNulls(size)
        }
    }

    override fun toString(): String {
        return "[name: ${name},price: ${price}$]"
    }

}